### DISCUSSION TITLE
Several improvements with the /sit

### DISCUSSION THREAD

**Discussion by NoltoxGit** [Zrips/CMI#6708] [Feature Requests]
Hello CMI Community!

Before using CMI I was using the [GSit](https://www.spigotmc.org/resources/gsit-modern-sit-seat-and-chair-lay-and-crawl-plugin-1-14-x-1-18-x.62325/) plugin, but when switching to CMI I noticed that it was missing quite a few things that could be added and also as other players have suggested. These are very funny features :)

Here is the list of additions I would like to see:

1. Command: `/sittoggle`
Usage: Allows to switch the sitting on the tiles/staircase automatically.

2. Command: `/sleep`
Usage : Allows you to make an animation as if you were [sleeping](https://imgur.com/XkwPgdM).

3. Command: `/spin`
Usage: A rather cosmetic animation that allows the player to [spin](https://imgur.com/fKWFObW) around. Similar to the trident but without moving.

4. Command: `/swim`
Usage : Put the player in [swimming](https://imgur.com/1y2uDkk) position.

- **_If you have other ideas for orders like this; put them below I will add them as time goes on!_**

**Comment by mrfloris** [comment]
I think a few are only possible with protocollib support, but yes, it would be cool to have these commands, i remember asking for them ages ago. So you have my upvote.

**Reply by NoltoxGit** [reply]
Noo, not ProtocolLib! Use packetevents instead

**Comment by CloudeLecaw** [comment]
ohh yes this would be a so great idea :D

well may also a little animation?
**/dance** ? :-)

**Comment by mercurialmusic** [comment]
Here 3 years later to upvote the sit toggle. Had several people in my servers requesting this lately.

**Reply by CloudeLecaw** [reply]
im allways for funny new things like that, that allow players to have more fun in kind if more options to interact. so yes...up it :D

**Comment by Zrips** [comment]
You can right click slabs, stairs or carpets to sit on them already as long as its enabled in config file and you have cmi.command.sit.stairs permission node, duno about toggle command, what exact purpose would it have here. Would it simply disable this automatic sitting?

**Reply by mercurialmusic** [reply]
> You can right click slabs, stairs or carpets to sit on them already as long as its enabled in config file and you have cmi.command.sit.stairs permission node, duno about toggle command, what exact purpose would it have here. Would it simply disable this automatic sitting?

For my part at least, it's strangely one of the more frequently requested things. Players have been complaining about mis-clicking on the slabs and being jumped to a position they did not intend to be in. I've done it myself a couple times and it can be a little annoying when unintentional. We considered adding our own toggle which negates or re-sets the permission for the player, but that option seems a little extreme, especially since players have a tendency to spam buttons--to me it sounds like a quick way to corrupt a database by letting players toy with permissions. But yeah, the toggle in this case would temporarily disable the auto-sit for that player.

**Reply by CloudeLecaw** [reply]
we also made this tiggle and added a delay into it at the end. this way players cant use it for a while like a cooldown, so a spam of it from them is useless.
we say "No player needs to toggle this in a frequently way, so its fine if he has the option each xx seconds once"
